cirq/ops/reversible_composite_gate.py

```diff
@@ -23,34 +23,26 @@
 
 
 class ReversibleCompositeGate(gate_features.CompositeGate,
-                              gate_features.ReversibleEffect,
                               metaclass=abc.ABCMeta):
     """A composite gate that gets decomposed into reversible gates."""
 
-    def __pow__(self, power):
+    def __pow__(self: TOriginal,
+                power) -> '_ReversedReversibleCompositeGate[TOriginal]':
```

Comment:
ReversedReversibleCompositeGate?  I guess it's okay, but it is a really goofy name.

Reply:
Ack. It's private, so it's fine if it's long-winded.